runtime/lua/lsp/callbacks.lua

```diff
@@ -0,0 +1,261 @@
+-- Implements the following default callbacks:
+--  textDocument/publishDiagnostics
+--  notification: textDocument/didOpen
+--  notification: textDocument/willSave
+--  TODO: textDocument/willSaveWaitUntil
+--  notification: textDocument/didSave
+--  notification: textDocument/didClose
+--  IN PROGRESS: textDocument/completion
+--  textDocument/hover
+--  TODO: textDocument/signatureHelp
+--  textDocument/references
+--  TODO: textDocument/documentHighlight
+--  TODO: textDocument/documentSymbol
+--  TODO: textDocument/formatting
+--  TODO: textDocument/rangeFormatting
+--  TODO: textDocument/onTypeFormatting
+--  textDocument/definition
+--  TODO: textDocument/codeAction
+--  TODO: textDocument/codeLens
+--  TODO: textDocument/documentLink
+--  TODO: textDocument/rename
+--
+--  TODO: completionItem/resolve
+--
+--  TODO: codeLens/resolve
+--
+--  TODO: documentLink/resolve
+
+local log = require('lsp.log')
+local util = require('neovim.util')
+local lsp_util = require('lsp.util')
+
+local protocol = require('lsp.protocol')
+local handle_completion = require('lsp.handle.completion')
+
+local error_callback = require('lsp.config.callbacks').error_callback
+
+local cb = {}
+cb.textDocument = {}
+
+cb.textDocument.publishDiagnostics = function(success, data)
```

Comment:
There seems to be no canonical way of changing/adding a notification handler, I guess `require('lsp.callbacks').callbacks.textDocument.publishDiagnostics = myhandler` works but seems fragile and clumsy and also it could make sense for multiple plugins to recieve the same notification,